### DISCUSSION TITLE
Training issues

### DISCUSSION THREAD

**Discussion by ZJ-CAI** [neonbjb/tortoise-tts#106] [General]
Hello Mr.Betker. I`d like to train this model but meet with a problem. 

I input the 3 mel-spectrograms of the target voice and the text, and then i get the output which is used to calculate loss with target.  For instance, the input is the mel_spectrograms of speaker Angies speechs(speechA, speechB & speechC) and a text of Angies speechD. The target would be the mel_spectrogram of  speaker Angie`s speechD.

Unfortunately, after the whole processes(autoregressive, diffusion, clvp, vocoder), the shape of the output is "torch.Size([1, 557056])". Whilst the shape of target mel_spectrogram is "torch.Size([1, 95520])". Thus, i am confused how to calculate the loss between the output and target? 

Besides, is there any solution to make their shapes be the same?

**Comment by neonbjb** [comment]
Hey,
You can't train Tortoise end to end. Each of the component models must be trained piecewise since they all have very different objective functions. It's a fairly involved, complicated process. I recommend you start by trying to train a small DALLE model and a diffusion model. These will get you the skills necessary to train Tortoise.

**Comment by ZJ-CAI** [comment]
Thank you for your reply. 
Does DALLE and diffusion model respectively refer to **https://github.com/lucidrains/DALLE-pytorch**  and    **https://github.com/openai/guided-diffusion** ?

**Comment by neonbjb** [comment]
Sure, those would work.
